docs/developer/releasing.md

```diff
@@ -0,0 +1,42 @@
+# Releasing
+
+## Output
+
+### Expected artifacts
+
+1. A container image of the shared cluster-api controller manager
+2. A git tag for providers to use
+
+### Artifact locations
+
+1. The container image is found in the registry `gcr.io/k8s-cluster-api` with an
+   image name of `cluster-api-controller` and a tag that matches the release
+   version. For example, in the `0.0.0-alpha.4` release, the container image
+   location is `gcr.io/k8s-cluster-api/cluster-api-controller:0.0.0-alpha.4`
+
+## Process
+
+1. Create a pull request that contains two changes:
+   1. Change [the cluster api controller manager image
```

Comment:
Could we get a `Makefile` target that can make these changes, and second target to undo them?

Reply:
I agree, this PR dance is by far the most onerous part of the release process.

Reply:
It reminds me of Maven releases 😄